### DISCUSSION TITLE
Simplest user input handling

### DISCUSSION THREAD

**Discussion by alexandra-petra** [python-telegram-bot/python-telegram-bot#3084] [Q&A]
Hi, I am writing a simple Telegram Bot which for testing words on flash-card principle. I give a user a word in one language, and ask him to write a translation in another language. For that I have a short list of ordered words in both languages. For that I

1. randomize words in first list
2. get index of the word
3. check if the input given by user == to the word with index in 2) in second list.
4.  If it's correct, send to user reply "Верно" + send sticker.

That's the part where I am stuck. I simply don't know how to store user input. I thought it was as straigtforward as "user_input_rus = update.message.text", but it seems I'm updating the message which is reserved for "Test 🇷🇺":, which is a button.

```
def button_choice(message):  
    if message.chat.type == "private":
    if message.text == "Test 🇷🇺":
        random_se = words_se[random.randint(0, len(words_se) - 1)]
        bot.send_message(message.chat.id, random_se)
        ind_se = words_se.index(random_se)
        rus_word = words_ru[ind_se] 
        bot.send_message(message.chat.id, "Skriv skriv detta ord på ryska:")
        user_input_rus = update.message.text  # here is the line for user input 
        if user_input_rus == rus_word:
            bot.send_message(message.chat.id, "Верно!")
            yes = open("/Users/alexandra/Documents/Bots_Telegram/Emil.tgs", 'rb')
            bot.send_sticker(message.chat.id, yes)
```

What happens now: Bot works, but when I type my word as a user, it doesn't proceed further, and nothing happens, so I suspect I mess it up in a user input line.

**Comment by Poolitzer** [comment]
> I thought it was as straigtforward as "user_input_rus = update.message.text", but it seems I'm updating the message which is reserved for "Test ru":, which is a button.

Yeah no, you don't really want this. This would mean your code just stops executing, so this breaks as soon as more then one person uses the bot. 

You probably want to use the ConversationHandler, have a look at https://github.com/python-telegram-bot/python-telegram-bot/tree/v13.x/examples#conversationbotpy
